map/extrapolation/extrapolator.cpp

```diff
@@ -0,0 +1,162 @@
+#include "map/extrapolation/extrapolator.hpp"
+
+#include "geometry/distance_on_sphere.hpp"
+
+#include "platform/platform.hpp"
+
+#include "base/logging.hpp"
+
+#include <chrono>
+#include <memory>
+
+namespace
+{
+uint64_t constexpr kMaxExtrapolationTimeMs = 1000;
+uint64_t constexpr kExtrapolationPeriodMs = 200;
+double constexpr kMaxExtrapolationSpeedMPerS = 120.0;
+
+double LinearExtrapolationOfOneParam(double param1, double param2, uint64_t timeBetweenPointsMs,
+                                     uint64_t timeAfterPoint2Ms)
+{
+  return (param2 - param1) * timeAfterPoint2Ms / timeBetweenPointsMs;
+}
+}  // namespace
+
+namespace position_extrapolator
+{
+using namespace std;
+
+location::GpsInfo LinearExtrapolation(location::GpsInfo const & point1, location::GpsInfo const & point2,
+                                      uint64_t timeAfterPoint2Ms)
+{
+  auto const timeBetweenPointsMs =
+      static_cast<uint64_t>((point2.m_timestamp - point1.m_timestamp) * 1000);
+
+  location::GpsInfo extrapolated = point2;
+
+  extrapolated.m_timestamp += timeAfterPoint2Ms;
+
+  extrapolated.m_longitude += LinearExtrapolationOfOneParam(
+      point1.m_longitude, point2.m_longitude, timeBetweenPointsMs, timeAfterPoint2Ms);
```

Comment:
Please pass the common params to a small helper class.

```
class LinearExtrapolator
{
public:
  LinearExtrapolator(uint64_t timeBetween, uint64_t timeAfter):
    m_timeBetween(timeBetween), m_timeAfter(timeAfter) {}

  double Extrapolate(double x1, double x2)
  {
    return x2 + (x2 - x1) / timeBetween * timeAfter;
  }

private:
  uint64_t m_timeBetween;
  uint64_t m_timeAfter;
};
```

Reply:
Done. It's looks much better with such helper class. Thanks.